src/sage/knots/link.py

```diff
@@ -2362,6 +2403,48 @@ def regions(self):
             regions.append(region)
         return regions
 
+    def remove_loops(self):
+        r"""
+        Return an ambient isotopic link in which all loops are removed.
+
+        EXAMPLES::
+
+          sage: b = BraidGroup(4)((3, 2, -1, -1))
+          sage: L = Link(b)
+          sage: L.remove_loops()
+          Link with 2 components represented by 2 crossings
+          sage: K4 = Link([[1, 7, 2, 6], [3, 1, 4, 8], [5, 5, 6, 4], [7, 3, 8, 2]])
+          sage: K3 = K4.remove_loops()
+          sage: K3.pd_code()
+          [[1, 7, 2, 4], [3, 1, 4, 8], [7, 3, 8, 2]]
+          sage: U = Link([[1, 2, 2, 1]])
+          sage: U.remove_loops()
+          Link with 1 component represented by 0 crossings
+        """
+        pd = self.pd_code()
+        from copy import copy
+        new_pd = [copy(cr) for cr in pd if len(set(cr)) == 4]
```

Comment:
```suggestion
        new_pd = [list(cr) for cr in pd if len(set(cr)) == 4]
```
This can be done to make a (shallow) copy of a list.

Reply:
Good to know!